### ISSUE TITLE
Test failure(s): core: LocaleProviderOSGiTest

### ISSUE THREAD

**Issue by maggu2810** [eclipse/smarthome#1731]
New introduced test breaks build.
Related to #1730

```text
Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.061 sec - in org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest
assert two items with same name can not be added(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0.014 sec
assert items are there once the factory gets added(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0.018 sec
assert items are added as group members on deferred creation(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0.006 sec
assert getItems returns item from registered ManagedItemProvider(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0.003 sec
assert removal returns old value(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0.003 sec
assert tags are stored and retrieve as well(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0.003 sec
updating existing item returns old value(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0.004 sec
assert remove recursively works(org.eclipse.smarthome.core.storage.ManagedItemProviderOSGiTest)  Time elapsed: 0.004 sec

Results :

Failed tests: 
  LocaleProviderOSGiTest.assert that default locale is provided if no locale has been configured:60 
Expected: is <de>
     but: was <de_DE>

Tests run: 162, Failures: 1, Errors: 0, Skipped: 1
```

**Comment by kaikreuzer** [comment]
@tomhoefer Please check!

**Comment by maggu2810** [comment]
Hm, I am confused...

I am inserted this code in front of the test that fails.

```java
System.out.println("Set GERMAN")
Locale.setDefault(Locale.GERMAN)
System.out.println("Locale.getDefault(): " + Locale.getDefault());
System.out.println("localeProvider.getLocale(): " + localeProvider.getLocale());
System.out.println("Set ENGLISH")
Locale.setDefault(Locale.ENGLISH)
System.out.println("Locale.getDefault(): " + Locale.getDefault());
System.out.println("localeProvider.getLocale(): " + localeProvider.getLocale());
```

The result is:

```text
Set GERMAN
Locale.getDefault(): de
localeProvider.getLocale(): de_DE
Set ENGLISH
Locale.getDefault(): en
localeProvider.getLocale(): de_DE
```

My system is using the english locale.

So I think there is the CM used by the locale provider (the only one that could inject de_DE -- I assume).

**Comment by maggu2810** [comment]
I added further debug messages to the locale provider.

```diff
diff --git a/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/i18n/LocaleProviderImpl.java b/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/i18n/LocaleProviderImpl.java
index 7fd1bfe..fb4868e 100644
--- a/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/i18n/LocaleProviderImpl.java
+++ b/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/i18n/LocaleProviderImpl.java
@@ -42,6 +42,8 @@ public class LocaleProviderImpl implements LocaleProvider {
         final String region = (String) config.get(REGION);
         final String variant = (String) config.get(VARIANT);
 
+        System.out.println(String.format("LocaleProvider modified: language=%s, script=%s, region=%s, variant=%s", language, script, region, variant));
+
         if (StringUtils.isEmpty(language)) {
             // if no language is set, skip new locale...
             logger.debug("No language set, keeping {} as locale", locale);
@@ -83,6 +85,7 @@ public class LocaleProviderImpl implements LocaleProvider {
 
     @Override
     public Locale getLocale() {
+        System.out.println(String.format("LocaleProvide getLocale: stored locale is: %s", locale));
         if (locale == null) {
             return Locale.getDefault();
         }
```

```text
Running org.eclipse.smarthome.core.internal.i18n.I18nProviderOSGiTest
LocaleProvider modified: language=de, script=null, region=DE, variant=null
LocaleProvide getLocale: stored locale is: de_DE
LocaleProvide getLocale: stored locale is: de_DE
LocaleProvide getLocale: stored locale is: de_DE
LocaleProvide getLocale: stored locale is: de_DE
LocaleProvider modified: language=de, script=null, region=DE, variant=null
LocaleProvide getLocale: stored locale is: de_DE
LocaleProvide getLocale: stored locale is: de_DE
LocaleProvide getLocale: stored locale is: de_DE
LocaleProvide getLocale: stored locale is: de_DE
LocaleProvider modified: language=de, script=null, region=DE, variant=null
LocaleProvider modified: language=de, script=null, region=DE, variant=null
LocaleProvide getLocale: stored locale is: de_DE
LocaleProvide getLocale: stored locale is: de_DE
LocaleProvide getLocale: stored locale is: de_DE
LocaleProvide getLocale: stored locale is: de_DE
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.049 sec - in org.eclipse.smarthome.core.internal.i18n.I18nProviderOSGiTest
assert that getText via bundle is working properly(org.eclipse.smarthome.core.internal.i18n.I18nProviderOSGiTest)  Time elapsed: 0.032 sec
assert that getText with arguments delegates properly to getText without arguments(org.eclipse.smarthome.core.internal.i18n.I18nProviderOSGiTest)  Time elapsed: 0.004 sec
assert that getText without bundle is working properly(org.eclipse.smarthome.core.internal.i18n.I18nProviderOSGiTest)  Time elapsed: 0.001 sec
assert that getText with arguments via bundle is working properly(org.eclipse.smarthome.core.internal.i18n.I18nProviderOSGiTest)  Time elapsed: 0.005 sec
Running org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest
LocaleProvide getLocale: stored locale is: de_DE
Set GERMAN
Locale.getDefault(): de
LocaleProvide getLocale: stored locale is: de_DE
localeProvider.getLocale(): de_DE
Set ENGLISH
Locale.getDefault(): en
LocaleProvide getLocale: stored locale is: de_DE
localeProvider.getLocale(): de_DE
LocaleProvide getLocale: stored locale is: de_DE
LocaleProvide getLocale: stored locale is: de_DE
LocaleProvider modified: language=de, script=Latn, region=DE, variant=null
LocaleProvide getLocale: stored locale is: de_DE_#Latn
LocaleProvider modified: language=ru, script=Cyrl, region=RU, variant=null
LocaleProvide getLocale: stored locale is: ru_RU_#Cyrl
Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.023 sec <<< FAILURE! - in org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest
assert that locale provider does not return null value(org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest)  Time elapsed: 0.002 sec
assert that default locale is provided if no locale has been configured(org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest)  Time elapsed: 0.011 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[de]> but was:<[en]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.junit.Assert$assertEquals$1.callStatic(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:53)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:157)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:169)
	at org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest.assert that default locale is provided if no locale has been configured(LocaleProviderOSGiTest.groovy:72)

assert that configured locale is provided(org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest)  Time elapsed: 0.004 sec
```

That is the "difficult" about the introduced relying on the system default locale, it is only working if no other one injects a default locale using CM.

**Comment by maggu2810** [comment]
The I18nProviderOSGiTest setup the locale provider in its setUp method.
It is not using the Locale.setDefault() method, but modifies the internal one.

You could change this one, to use also Locale.setDefault() but I still think it is very confusing because the order of test execution etc. could change result. So every test needs to use Locale.setDefault() or no one or we have to check the execution order (for the ones running in the same env.).

**Comment by tomhoefer** [comment]
will fix it

**Comment by maggu2810** [comment]
@kaikreuzer Should we revert #1730 as long as it is fixed? Every build on Hudson will fail as the test will never succeed.

**Comment by tomhoefer** [comment]
isnt it fixed by my last change for #1732 ?

**Comment by maggu2810** [comment]
As Github does not send a notification (it seems they started with, but at least not if you rebase the branch instead of adding commits), you should add a notification, after you have added a new commit.

**Comment by maggu2810** [comment]
I updated a workstation to Neon.
So I have done a fresh installation and checkout.
After that I started the first "mvn clean install" for the newly checked out repo.

The build does not succeed:

```text
Tests run: 5, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 10.384 sec <<< FAILURE! - in org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest
assert that default locale is provided after locale has been cleared(org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest)  Time elapsed: 0.098 sec
assert that locale provider does not return null value(org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest)  Time elapsed: 0.014 sec
assert that default locale is provided if no locale has been configured(org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest)  Time elapsed: 0.015 sec
assert that configured locale is provided(org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest)  Time elapsed: 0.143 sec
assert that locale can be updated using directly the modified operation(org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest)  Time elapsed: 10.104 sec  <<< FAILURE!
java.lang.AssertionError: 
Expected: is "es"
     but: was "en"
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:865)
	at org.junit.Assert.assertThat(Assert.java:832)
	at org.junit.Assert$assertThat.callStatic(Unknown Source)
	at org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest$_assertLocale_closure5.doCall(LocaleProviderOSGiTest.groovy:133)
	at org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest$_assertLocale_closure5.doCall(LocaleProviderOSGiTest.groovy)
	at sun.reflect.GeneratedMethodAccessor8.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:278)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1016)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:39)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:112)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:181)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:141)
	at org.eclipse.smarthome.core.internal.i18n.LocaleProviderOSGiTest.assertLocale(LocaleProviderOSGiTest.groovy:130)
```

**Comment by maggu2810** [comment]
I assume you should a waitForAssert here:
https://github.com/qivicon/smarthome/blob/aa34daa0deb1c75114be15936e5de1a458d73c0a/bundles/core/org.eclipse.smarthome.core.test/src/test/groovy/org/eclipse/smarthome/core/internal/i18n/LocaleProviderOSGiTest.groovy#L94

**Comment by maggu2810** [comment]
Ignore my comment :wine_glass:

**Comment by tomhoefer** [comment]
should be fixed bx #1750

**Comment by maggu2810** [comment]
Thanks
